source/ice_api.c

```diff
@@ -0,0 +1,1676 @@
+    uint64_t controlledAgentCandidatePri = pIceCandidatePair->pRemote->priority;
+
+    if( pIceCandidatePair != NULL )
+    {
+        if( isLocalControlling == 0 )
+        {
+            controllingAgentCandidatePri = controlledAgentCandidatePri;
+            controlledAgentCandidatePri = pIceCandidatePair->pLocal->priority;
+        }
+
+        return( ( ( uint64_t ) 1 << 32 ) * ( controllingAgentCandidatePri >= controlledAgentCandidatePri ? controlledAgentCandidatePri : controllingAgentCandidatePri ) +
+                2 * ( controllingAgentCandidatePri >= controlledAgentCandidatePri ? controllingAgentCandidatePri : controlledAgentCandidatePri ) + ( controllingAgentCandidatePri > controlledAgentCandidatePri ? 1 : 0 ) );
+    }
+    else
+    {
+        return 0;
+    }
+}
+/*------------------------------------------------------------------------------------------------------------------*/
+
+/* Ice_CreateTransactionIdStore - Creates the Transaction ID Store. */
+
+static IceResult_t Ice_CreateTransactionIdStore( uint32_t maxIdCount,
+                                                 TransactionIdStore_t * pTransactionIdStore )
+{
+    IceResult_t retStatus = ICE_RESULT_OK;
+
+    if( ( pTransactionIdStore == NULL ) || ( maxIdCount > ICE_MAX_STORED_TRANSACTION_ID_COUNT ) || ( maxIdCount < 0 ) )
+    {
+        retStatus = ICE_RESULT_BAD_PARAM;
+    }
+
+    if( retStatus == ICE_RESULT_OK )
+    {
+        pTransactionIdStore->pTransactionIds = ( uint8_t * ) ( pTransactionIdStore + 1 );
+        pTransactionIdStore->maxTransactionIdsCount = maxIdCount;
+    }
+
+    return retStatus;
+}
+/*------------------------------------------------------------------------------------------------------------------*/
+
+/* Ice_TransactionIdStoreInsert - Inserts the Transaction in the IceAgent Transaction ID Store. */
+
+static void Ice_TransactionIdStoreInsert( TransactionIdStore_t * pTransactionIdStore,
+                                          uint8_t * pTransactionId )
+{
+    uint8_t * storeLocation = NULL;
+    uint32_t transactionIDCount;
+
+    if( pTransactionIdStore != NULL )
+    {
+        storeLocation = pTransactionIdStore->pTransactionIds + ( ( pTransactionIdStore->nextTransactionIdIndex % pTransactionIdStore->maxTransactionIdsCount ) * STUN_HEADER_TRANSACTION_ID_LENGTH );
+        memcpy( storeLocation,
+                pTransactionId,
+                STUN_HEADER_TRANSACTION_ID_LENGTH );
+
+        pTransactionIdStore->nextTransactionIdIndex = ( pTransactionIdStore->nextTransactionIdIndex + 1 ) % pTransactionIdStore->maxTransactionIdsCount;
+
+        if( pTransactionIdStore->nextTransactionIdIndex == pTransactionIdStore->earliestTransactionIdIndex )
+        {
+            pTransactionIdStore->earliestTransactionIdIndex =
+                ( pTransactionIdStore->earliestTransactionIdIndex + 1 ) % pTransactionIdStore->maxTransactionIdsCount;
+            return;
```

Comment:
We are following this rule of return at the end. I think that needs to be consistent everywhere.

Reply:
updated.